src/pages/City/CityPage.tsx

```diff
@@ -15,19 +16,18 @@ const CityPage = () => {
   const { data, isLoading } = Weather.useLocation(city);
   const { state }: NavigateOptions = useLocation();
 
-  const renderCityName = () => {
-    if (isLoading) {
-      return <CityNameLoader />;
-    }
-    if (data) {
-      return <CityName>{state?.cityName || data.location.name}</CityName>;
-    }
-  };
-
   return (
     <PageContentContainer data-testid={CityPageTestIds.Container}>
-      {renderCityName()}
+      <CityNameWidget
+        loading={isLoading}
+        cityName={data !== undefined ? state?.cityName || data.location.name : undefined}
+      />
       <WidgetWrapper data={data} isLoading={isLoading} data-testid={CityPageTestIds.Widgets} />
+      <NearestCitiesTitle>nearest</NearestCitiesTitle>
+      <NearestCitiesWeatherWidget
+        loading={isLoading}
+        cityName={data !== undefined ? state?.cityName || data.location.name : undefined}
```

Comment:
**suggestion** I believe we agreed to stick to `WeatherApi` location name. Giving cityName with state from overpass would be problematic in couple areas like sending someone URL link or reloading page